### PR TITLE
extendend the geo plugin leaflet.js to configure overlays | adding chmaps

### PR DESCRIPTION
**Pull request by l00mi**
added **chmaps** to leaflet.js

### CODE REVIEW

**Comment by l00mi** [comment]
@LaurensRietveld do you think this can go in like this until the end of the week?

**Comment by LaurensRietveld** [comment]
Thanks! And sure, I can add this. Could  you remove the build files and such from the PR? Those are automatically rebuild when I publish the new version

**Comment by l00mi** [comment]
there you go!

File: gulp/makeCss.js [hunk 1]
```diff
@@ -62,3 +62,7 @@ gulp.task("copyCssDeps", function() {
     )
     .pipe(gulp.dest("./src/scss/cssIncludes"));
 });
+gulp.task('copyCssImages', function() {
+   gulp.src('./node_modules/leaflet/dist/images/*.png')
+   .pipe(gulp.dest(paths.bundleDir+'/images'));
+});
```
Comment (LaurensRietveld) [inline gulp/makeCss.js:68]: not sure about the css images. Is there a need to copy them? Re-using the YASR lib as dep would require you to serve these images as well from that specific path

**Comment by LaurensRietveld** [comment]
the map looks good to me. Could you check the comment about the buildCss file? afaik we can ignore this one right?

File: gulp/makeCss.js (hunk 1 above)
Comment (l00mi) [reply gulp/makeCss.js:68]: yes, we need the layers icon for the overlay extension

Comment (LaurensRietveld) [reply gulp/makeCss.js:68]: instead, couldn't you base encode the image you need, and put that in the leaflet scss (`/src/leaflet.scss/`)? e.g., if it's the layer-toggle icon you'd like, you could add something like this:
```
.leaflet-control-layers-toggle {
background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABoAAAAaCAQAAAADQ4RFAAACf0lEQVR4AY1UM3gkARTePdvdoTxXKc+qTl3aU5U6b2Kbkz3Gtq3Zw6ziLGNPzrYx7946Tr6/ee/XeCQ4D3ykPtL5tHno4n0d/h3+xfuWHGLX81cn7r0iTNzjr7LrlxCqPtkbTQEHeqOrTy4Yyt3VCi/IOB0v7rVC7q45Q3Gr5K6jt+3Gl5nCoDD4MtO+j96Wu8atmhGqcNGHObuf8OM/x3AMx38+4Z2sPqzCxRFK2aF2e5Jol56XTLyggAMTL56XOMoS1W4pOyjUcGGQdZxU6qRh7B9Zp+PfpOFlqt0zyDZckPi1ttmIp03jX8gyJ8a/PG2yutpS/Vol7peZIbZcKBAEEheEIAgFbDkz5H6Zrkm2hVWGiXKiF4Ycw0RWKdtC16Q7qe3X4iOMxruonzegJzWaXFrU9utOSsLUmrc0YjeWYjCW4PDMADElpJSSQ0vQvA1Tm6/JlKnqFs1EGyZiFCqnRZTEJJJiKRYzVYzJck2Rm6P4iH+cmSY0YzimYa8l0EtTODFWhcMIMVqdsI2uiTvKmTisIDHJ3od5GILVhBCarCfVRmo4uTjkhrhzkiBV7SsaqS+TzrzM1qpGGUFt28pIySQHR6h7F6KSwGWm97ay+Z+ZqMcEjEWebE7wxCSQwpkhJqoZA5ivCdZDjJepuJ9IQjGGUmuXJdBFUygxVqVsxFsLMbDe8ZbDYVCGKxs+W080max1hFCarCfV+C1KATwcnvE9gRRuMP2prdbWGowm1KB1y+zwMMENkM755cJ2yPDtqhTI6ED1M/82yIDtC/4j4BijjeObflpO9I9MwXTCsSX8jWAFeHr05WoLTJ5G8IQVS/7vwR6ohirYM7f6HzYpogfS3R2OAAAAAElFTkSuQmCC')
}
```

Comment (l00mi) [reply gulp/makeCss.js:68]: further, the path is correct because it is relative to the CSS file, that is why I co-located it to the makeCssImages.

Comment (LaurensRietveld) [reply gulp/makeCss.js:68]: But if I include YASGUI (with YASR as dep) in another library, then I'd need to do more than just pull in the js and css as dep. Now I'd also need to make sure I've downloaded the corresponding img assets, and host them correctly. That is an unnecessary burden to put on other developers, as it's something we can avoid with little cost (the base encoded image is only ~300 bytes larger than the actual png)

Comment (l00mi) [reply gulp/makeCss.js:68]: So you propose to include it directly into the CSS?

Comment (LaurensRietveld) [reply gulp/makeCss.js:68]: jup, just for those icon's that are needed (afaik only that toggle icon right?)

Comment (l00mi) [reply gulp/makeCss.js:68]: Yes the toggle icon is the only one I really need, there are only three if I remember correctly so I though whatever. Can you hint me where to add this manually?

Comment (LaurensRietveld) [reply gulp/makeCss.js:68]: it looks like you could revert the gulp makeCss changes, and update the scss file here instead:
https://github.com/OpenTriply/YASGUI.YASR/blob/gh-pages/src/leaflet.scss#L5 (with background-image snippet from above)

Comment (l00mi) [reply gulp/makeCss.js:68]: okay, done

**Comment by l00mi** [comment]
also added one more option to style the map in more detail defaultStyle

**Comment by l00mi** [comment]
please as soon as you can merge this, to also publish a new version on npm

**Comment by LaurensRietveld** [comment]
you forgot to remove the copy-img task in gulp. I merged your PR and removed this part of the code. new version is available as `2.11.8`

**Comment by l00mi** [comment]
perfect timing! thank you

**Comment by l00mi** [comment]
could you also bump yasgui please (-, ?

**Comment by LaurensRietveld** [comment]
jup, see yasgui@2.6.11

**Comment by l00mi** [comment]
great, thank you!

**Comment by l00mi** [comment]
okay, looks good so far, only the icon did not survive if we use the yasgui package, any ideas?

https://web-ld-trifid.dev.bgdi.ch/sparql/#query=+PREFIX+rdf%3A+%3Chttp%3A%2F%2Fwww.w3.org%2F1999%2F02%2F22-rdf-syntax-ns%23%3E%0APREFIX+rdfs%3A+%3Chttp%3A%2F%2Fwww.w3.org%2F2000%2F01%2Frdf-schema%23%3E%0APREFIX+wdt%3A+%3Chttp%3A%2F%2Fwww.wikidata.org%2Fprop%2Fdirect%2F%3E%0APREFIX+wikibase%3A+%3Chttp%3A%2F%2Fwikiba.se%2Fontology%23%3E%0APREFIX+skos%3A+%3Chttp%3A%2F%2Fwww.w3.org%2F2004%2F02%2Fskos%2Fcore%23%3E%0APREFIX+xsd%3A+%3Chttp%3A%2F%2Fwww.w3.org%2F2001%2FXMLSchema%23%3E%0APREFIX+wd%3A+%3Chttp%3A%2F%2Fwww.wikidata.org%2Fentity%2F%3E%0APREFIX+fn%3A+%3Chttp%3A%2F%2Fwww.w3.org%2F2005%2Fxpath-functions%23%3E%0APREFIX+geo%3A+%3Chttp%3A%2F%2Fwww.opengis.net%2Font%2Fgeosparql%23%3E%0A%0ASELECT+%3FMunicipality+%3FName++%3FCoords+%3FCoordsLabel+WHERE+%7B%0A++++%3FMunicipality+a+%3Chttp%3A%2F%2Fwww.geonames.org%2Fontology%23A.ADM3%3E+.+%23municipality%0A++++%3FMunicipality+%3Chttp%3A%2F%2Fschema.org%2Fname%3E+%3FName+.%09%0A++++%3FMunicipality+%3Chttp%3A%2F%2Fwww.geonames.org%2Fontology%23parentADM1%3E+%3FInCanton+.%0A++++%3FInCanton+%3Chttp%3A%2F%2Fschema.org%2Fname%3E+%3FCantonName+.%0A++++%3FMunicipality+%3Chttp%3A%2F%2Fpurl.org%2Fdc%2Fterms%2Fissued%3E+%3FDate+.%0A++++%3FMunicipality+%3Chttps%3A%2F%2Fld.geo.admin.ch%2Fdef%2FbfsNumber%3E+%3FbfsNumber+.+%23connect+to+the+%3FbfsNumber+found+in+wikidata.%0A++++%3FMunicipality+geo%3AhasGeometry+%3FGeometry+.%0A++++%3FGeometry+geo%3AasWKT+%3FCoords+.%0A++%23Filter+for+Municipalities+-+Version+2017%0A++FILTER+(%3FDate+%3E%3D+%222017-01-01%22%5E%5Exsd%3Adate)%0A++%23FILTER+for+Municiaplities+in+Canton+Vaud.+Change+the+name+for+other+Cantons+(https%3A%2F%2Fld.geo.admin.ch%2Fboundaries%2Fcanton%2F).%0A++FILTER+(%3FCantonName+%3D+%22Vaud%22)%0A++%0A++%7B%0A++++SELECT+DISTINCT+(xsd%3Ainteger(%3Fbfs)+AS+%3FbfsNumber)+%3FWikidataURI+%3FCoordsLabel+WHERE+%7B%0A+%09%09SERVICE+%3Chttps%3A%2F%2Fquery.wikidata.org%2Fbigdata%2Fnamespace%2Fwdq%2Fsparql%3E%0A%09%09%7B%0A++++++++%09%3FWikidataURI+wdt%3AP771+%3Fbfs+.+%23look+for+bfs+number+in+WD%0A++++++++++++%3FWikidataURI+wdt%3AP31+wd%3AQ70208+.+%23municipality+of+Switzerland%0A++++++++%09%3FWikidataURI+wdt%3AP473+%3FCoordsLabel+.+%0A++++%09%7D%0A++%09%7D%0A++%7D%0A%7D%0A&contentTypeConstruct=text%2Fturtle&contentTypeSelect=application%2Fsparql-results%2Bjson&endpoint=https%3A%2F%2Fld.geo.admin.ch%2Fquery&requestMethod=POST&tabTitle=Query&headers=%7B%7D&outputFormat=leaflet

**Comment by LaurensRietveld** [comment]
This is _probably_ due to some merge issues: while you were busy with the pr, I added support to using a scoped version of leaflet (i.e., you can pass your leaflet dep as argument using `YASR.plugins.leaflet.defaults.L`). The layer-part of your code used the global leaflet ref, where the rest uses the scoped leaflet ref. This is changed now (https://github.com/OpenTriply/YASGUI.YASR/commit/82c142f6f340a78cdbd655e1283af2a64a4c24e6).

YASR/YASGUI versions are upped as well, so you can try again

**Comment by l00mi** [comment]
works like a charm, thank you very much, if i find some time I will document the added functionality in 2-3 sentences

**Comment by LaurensRietveld** [comment]
good to hear!

**Pull request merged**